gateway-ha/src/main/java/io/trino/client/ServerInfo.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.client;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.Duration;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.Boolean.parseBoolean;
+import static java.util.Objects.requireNonNull;
+
+// based on https://github.com/trinodb/trino/blob/f83854687229df1ae3325249e7bcd29f03d0943c/client/trino-client/src/main/java/io/trino/client/ServerInfo.java#L28
+public class ServerInfo
+{
+    private final NodeVersion nodeVersion;
+    private final String environment;
+    private final boolean coordinator;
+    private final boolean starting;
+
+    // optional to maintain compatibility with older servers
+    private final Optional<Duration> uptime;
+
+    @JsonCreator
+    public ServerInfo(
+            @JsonProperty("nodeVersion") NodeVersion nodeVersion,
+            @JsonProperty("environment") String environment,
+            @JsonProperty("coordinator") boolean coordinator,
+            @JsonProperty("starting") String starting,
+            @JsonProperty("uptime") Optional<Duration> uptime)
+    {
+        this.nodeVersion = requireNonNull(nodeVersion, "nodeVersion is null");
+        this.environment = requireNonNull(environment, "environment is null");
+        this.coordinator = coordinator;
+        this.starting = parseBoolean(requireNonNull(starting, "starting is null"));
```

Comment:
It would be nice to leave code comment explaining why using String instead of boolean.